CONTRIBUTING.md

```diff
@@ -25,6 +25,52 @@ The following information is most helpful:
 * **Pointers** — If you have a hunch or just don't want to bother with a pull request,
 please point me in the right direction (e.g. line numbers) to fix it.
 
+## Development, Builds and Testing
+
+### Project Layout
+
+* **package.json** — Metadata about the project, including details for the extensions browser, dependencies, build scripts, commands and user configuration
+* **README.md** — User-facing documetnation for the extension
+* **CHANGELOG.md** — Version history and changes
+* **CONTRIBUTING.md** — This file
+* **LICENSE** — License information about the project
+* **NOTICE.txt** — License information about third party packages
+* **client** — Main extension folder
+    * **extension.ts** — Code insertion point.  Contains initialization code and defines commands callable by user.
+    * **marklogicClient.ts** — Interface to MarkLogic [Node Client API](https://github.com/marklogic/node-client-api) package, which is used to communicate with ML Server.
+    * **marklogic-types.d.ts** — TypeScript type declarations used by the package.
+* **server** — Language server extension folder (provides syntax highlighting; based on vscode-languageserver extension)
+    * **package.json** — As the language server is technically a separate package, this defines the dependencies for it
+    * **server.ts** — Code insertion point.  Contains initialization code for language server.
+
+### Building and Testing Project
+
+Building the project requires [Node.js](https://nodejs.org/) to be installed on your local machine
+
+First, dev dependencies must be installed.  Run `npm install` in both this folder, and in the `./server` folder.
```

Comment:
I like using the actual statements to clarify stuff like this when multiple folders are involved - e.g.:

    npm install
    cd server
    npm install
    cd ..

That makes it clear too that `npm install` doesn't need to be run from within the "client" folder (I'm assuming that's the case).

Reply:
It is.  The install should be run from the same folder as the package files, which are in root and server respectively.  I'll make that more explicit